force-app/main/default/classes/P1.cls

```diff
@@ -1,45 +1,71 @@
-public class P1 {
-    /**
-     * Return the nth fiboacci number in the fibonacci sequence.
-     * It should start 1,1,2,3...
-     * */
-    public static Integer fibonacci(Integer n) {
-        return 0;
-    }
+public with sharing class P1 {
+	/*** complete* */ 
+	public static Integer fibonacci(Integer n) {       
+        if(n<0){
+            return null;
+        }
+        Integer fib0=0,fib1=1,fib2=1;
+        for(Integer i=1;i<n;++i){    
+ 			fib2=fib0+fib1;        
+  			fib0=fib1;    
+  			fib1=fib2;    
+ 		}    
+  		return fib2; 
+	}
     
-    /**
-     * Given a string, return its backwards order.
-     * */
+    //*** complete */
     public static String reverseString(String s) {
-        return '';
+    	return s.reverse();
     }
     
-    /**
-     * Use recursion to compute the factorial of a given number.
-     * */
+    //*** complete */
     public static Integer factorial(Integer n) {
-        return 0;
+        if(n==0){
+            return 1;
+        }
+        else{
+        	return(n*factorial(n-1));
+    	}
     }
-    
-    /**
-     * Given a string, determine if it is a palindrome.
-     * */
+
+    //*** complete */    
     public static Boolean isPalindrome(String s) {
-        return false;
+        return s==s.reverse();
     }
     
-    /**
-     *  Given a state, return the accounts in that state.
-     *  I only want the accounts that have a contact who works at the same place as the account.
-     * */
+    
+    //*** complete */
     public static List<Account> getAccountsForAGivenState(State s) {
-        return new List<Account>();
+        String state = s.name();
+        List<Contact> Contacts = [SELECT MailingStreet FROM Contact WHERE AccountAddMatch__c = TRUE AND MailingState = :state ];
+        List<Account> Accounts = [SELECT Name, ID, BillingStreet FROM Account WHERE BillingState= :state]; 
+        List<Account> returnList = new List<Account>();
+        for (Account Account:Accounts){
+            for(Contact Contact:Contacts){
+                if (Account.Billingstreet==Contact.Mailingstreet){
```

Comment:
Shouldn't you also be checking if (Contact.AccountId == Account.Id)? On that, does this compile? The reason I ask is because you use _Account Account_ and _Contact Contact_ inside of your for loops, respectively, and you can't use keywords as variable names.